src/main/webapp/app/exam/participate/exam-navigation-bar/exam-navigation-bar.component.ts

```diff
@@ -27,9 +30,31 @@ export class ExamNavigationBarComponent implements OnInit {
 
     icon: IconProp;
 
-    constructor(private layoutService: LayoutService) {}
+    exerciseIdToNavigateTo: number;
+    subscriptionToExerciseNavigation: Subscription;
+
+    exerciseIdToUpdate: number;
+    exerciseProblemStatement: string;
+    subscriptionToLiveExamExerciseUpdates: Subscription;
+
+    constructor(
+        private layoutService: LayoutService,
+        private examExercisesNavigationService: ExamExercisesNavigationService,
+        private liveExamExerciseUpdateService: LiveExamExerciseUpdateService,
+    ) {}
 
     ngOnInit(): void {
+        this.subscriptionToExerciseNavigation = this.examExercisesNavigationService.currentExerciseIdToNavigateTo.subscribe((exerciseId) => {
+            this.exerciseIdToNavigateTo = exerciseId;
+            this.changeExerciseById(exerciseId);
+        });
+
+        this.subscriptionToLiveExamExerciseUpdates = this.liveExamExerciseUpdateService.currentExerciseIdAndProblemStatement.subscribe((update) => {
+            this.exerciseIdToUpdate = update['exerciseId'];
```

Comment:
If you implement my suggestion regarding adding an interface for this observable, you can directly access the fields here without using string keys...
```typescript
this.exerciseIdToUpdate = update.exerciseId;
```

Reply:
Great, thank you for your advice. It works just as you have described it.